File: backend/pos/src/main/java/com/grupo1/pos/model/DetalleVenta.java
package com.grupo1.pos.model;
import jakarta.persistence.*;

@Entity
@Table(name = "detalle_ventas")
public class DetalleVenta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "venta_id", nullable = false)
    private Venta venta;

    @ManyToOne
    @JoinColumn(name = "producto_id", nullable = false)
    private Producto producto;

    @Column(nullable = false)
    private Integer cantidad;

    @Column(nullable = false)
    private Double subtotal;


    // Getters and setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Venta getVenta() {
        return venta;
    }

    public void setVenta(Venta venta) {
        this.venta = venta;
    }

    public Producto getProducto() {
        return producto;
    }

    public void setProducto(Producto producto) {
        this.producto = producto;
    }

    public Integer getCantidad() {
        return cantidad;
    }

    public void setCantidad(Integer cantidad) {
        this.cantidad = cantidad;
    }

    public Double getSubtotal() {
        return subtotal;
    }

    public void setSubtotal(Double subtotal) {
        this.subtotal = subtotal;
    }
    public double getIva10(){
        return getSubtotal()/10;
    }
    public double getIva5(){
        return getSubtotal()/11;
    }

}

File: frontend/Sistema-pos/src/app/entidades/DetalleVenta.ts
export class DetalleVenta {
  id: number;          // ID del detalle de la venta
  cantidad: number;    // Cantidad del producto vendido
  subtotal: number;    // Subtotal (cantidad * precio del producto)
  producto_id: number; // ID del producto relacionado
  venta_id: number;    // ID de la venta a la que pertenece el detalle

  constructor(id: number, cantidad: number, subtotal: number, producto_id: number, venta_id: number) {
    this.id = id;
    this.cantidad = cantidad;
    this.subtotal = subtotal;
    this.producto_id = producto_id;
    this.venta_id = venta_id;
  }
}